### PR TITLE
Issues/22 ndjson support

### LINKED ISSUE #22: Improve mimetype check to support application/x-ndjson
Currently, Apache Tika does not support detection of the `application/x-ndjson` mimetype.  This mimetype is used by the TORCH tool to provide exports of MII CDS datasets for the data-transfer and data-sharing process. Therefore, the MimetypeHelper should support detection and check of this mimetype correctly.

### PR DESCRIPTION
**Pull request by wetret**
resolves #22

### CODE REVIEW

File: src/main/java/de/medizininformatik_initiative/processes/common/mimetype/CombinedDetectors.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+	public CombinedDetectors(List<Detector> detectors)
+	{
+		if (detectors != null && !detectors.isEmpty())
+			this.detectors.addAll(detectors);
+
+		if (this.detectors.isEmpty())
+			throw new RuntimeException("no detectors supplied");
+	}
+
+
```
Comment (hhund) [inline src/main/java/de/medizininformatik_initiative/processes/common/mimetype/CombinedDetectors.java:35]:
```suggestion
```

**Review by hhund — CHANGES_REQUESTED** [review]
Behavior of `NdJsonDetector` `detect` method should be improved. See line by line comments.

File: src/main/java/de/medizininformatik_initiative/processes/common/mimetype/NdJsonDetector.java [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+	 *            that should be checked for <code>application/x-ndjson</code>, not <code>null</code>
+	 * @param metadata
+	 *            will override property <code>Content-Type</code> as checks are line based, not <code>null</code>
+	 * @return <code>application/x-ndjson</code> if all lines match <code>application/json</code>,
+	 *         <code>application/x-empty</code> otherwise
+	 * @throws IOException
+	 *             if lines cannot be read from provided inputStream
+	 */
+	@Override
+	public MediaType detect(InputStream inputStream, Metadata metadata) throws IOException
```
[the remark this answers is not in the document]
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/processes/common/mimetype/NdJsonDetector.java:43]: implemented in 836f5b8

File: src/main/java/de/medizininformatik_initiative/processes/common/mimetype/NdJsonDetector.java [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+	 *
+	 * @param inputStream
+	 *            that should be checked for <code>application/x-ndjson</code>, not <code>null</code>
+	 * @param metadata
+	 *            will override property <code>Content-Type</code> as checks are line based, not <code>null</code>
+	 * @return <code>application/x-ndjson</code> if all lines match <code>application/json</code>,
+	 *         <code>application/x-empty</code> otherwise
+	 * @throws IOException
+	 *             if lines cannot be read from provided inputStream
+	 */
```
[the remark this answers is not in the document]
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/processes/common/mimetype/NdJsonDetector.java]: implemented in 836f5b8

File: src/main/java/de/medizininformatik_initiative/processes/common/mimetype/CombinedDetectors.java (hunk 1 above)
Comment (wetret) [reply src/main/java/de/medizininformatik_initiative/processes/common/mimetype/CombinedDetectors.java]: implemented in 836f5b8

**Pull request merged**